infrastructure/infrastructure.tf

```diff
@@ -40,26 +49,66 @@ provider "aws" {
 # Resources
 #############################################
 
-data "aws_ssm_parameter" "results_bucket_arn" {
-  name = "/${var.app_name}/${terraform.workspace}/s3/results/arn"
+locals {
+  # When a build is done as a user locally, or when building a stage e.g. dev/qa/prod we use
+  # the workspace name e.g. progers or dev
+  # When the circle ci build is run we override the var.ssm_source_stage to explicitly tell it
+  # to use the resources in dev. Change
+  ssm_source_stage = "${var.ssm_source_stage == "DEFAULT" ? terraform.workspace : var.ssm_source_stage}"
 }
 
 module "docker_image" {
-  source = "docker_image"
-  app_name = "${var.app_name}"
-  task_name = "${var.task_name}"
-  results_bucket_arn = "${data.aws_ssm_parameter.results_bucket_arn.value}"
+  source             = "docker_image"
+  app_name           = "${var.app_name}"
+  task_name          = "${var.task_name}"
+  results_bucket_arn = "${module.nmap_task.results_bucket_arn}"
+  results_bucket_id  = "${module.nmap_task.results_bucket_id}"
+  ssm_source_stage   = "${local.ssm_source_stage}"
+}
+
+data "aws_ssm_parameter" "vpc_id" {
+  name = "/${var.app_name}/${local.ssm_source_stage}/vpc/id"
+}
+
+resource "aws_security_group" "outbound_only" {
```

Comment:
use the security group in task execution

Reply:
done